### ISSUE TITLE
ClassCastException when calling LDClient.boolVariation

### ISSUE THREAD

**Issue by jclaudino** [launchdarkly/android-client#50]
Hi, we are getting reports of this crash happening to our customers, however we can't reproduce it ourselves. It doesn't seem related to the boolean value, rather the `flagVersion` is somehow stored as a float when it is expecting a string. 

```
java.lang.ClassCastException: java.lang.Float cannot be cast to java.lang.String
        at android.app.SharedPreferencesImpl.getString(SharedPreferencesImpl.java:255)
        at com.launchdarkly.android.response.BaseUserSharedPreferences.getValueAsJsonObject(BaseUserSharedPreferences.java:37)
        at com.launchdarkly.android.response.BaseUserSharedPreferences.extractValueFromPreferences(BaseUserSharedPreferences.java:27)
        at com.launchdarkly.android.response.UserFlagResponseSharedPreferences.getStoredFlagVersion(UserFlagResponseSharedPreferences.java:77)
        at com.launchdarkly.android.response.UserFlagResponseSharedPreferences.getVersionForEvents(UserFlagResponseSharedPreferences.java:153)
        at com.launchdarkly.android.LDClient.boolVariation(LDClient.java:326)
```

**Comment by brian98point6** [comment]
We are seeing the same exception signature, but calling LDClient.stringVariation. So far also unable to reproduce

**Comment by brian98point6** [comment]
Looks like this may be because of the 2.3.2 -> 2.4.0 upgrade. Version used to be stored as a float, and now it's stored a string:
https://github.com/launchdarkly/android-client/commit/ca1bb66fd62c742e779f08df31f1a190f77db9dd#diff-ac9a2d571d0be013d145404a5cb7804fL42

Returning users will have a float in their shared preferences, and with LDClient v2.4.0, the app will try to parse it as a string, causing this exception.

**Comment by brian98point6** [comment]
I still don't have a repro, so I'm not 100% sure of this...

**Comment by brian98point6** [comment]
I have a repro:
1. Install an app with LD v2.3.2 on it
2. Run it
3. Close the app
4. Turn on airplane mode (or a slow network might work)
5. Install the same app with LD v2.4.0 on it (don't uninstall or clear storage)
6. Run it
7. Observe the float to string exception

**Comment by arun251** [comment]
Thanks @jclaudino and @brian98point6 , we have a fix pending a release later tody.

**Comment by arun251** [comment]
This issue has been fixed in release 2.5.1.
